public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -353,7 +335,7 @@ export class LinkSrv implements LinkService {
   }
 }
 
-let singleton: LinkService;
+let singleton: LinkService = new LinkSrv();
```

Comment:
I see in few places you create the service on a module level. This will result in the service creation as soon as the module is imported. Is there sth that blocks us from creating a new instance in `getXXXSrv` if the `singleton` is undefined? We do that i.e. in `getTimeSrv`

Reply:
no that is a better pattern

Reply:
no that is a better pattern

Reply:
great, thanks